### PR TITLE
Make default bucket count thresholds immutable

### PR DESCRIPTION
**Pull request by benwtrent**
I found while debugging https://github.com/elastic/elasticsearch/issues/92058 that I had to double check that our default bucket values (which are static objects) are never mutated. 

This should be enforced by the compiler. So this adds a new `ConstantBucketCountThresholds` to be used in our `static final` `DEFAULT_BUCKET_COUNT_THRESHOLDS` objects.

This is a minor refactor, but gives me peace of mind :)

### CODE REVIEW

**Review by not-napoleon — APPROVED** [review]
LGTM

File: server/src/main/java/org/elasticsearch/search/aggregations/bucket/terms/TermsAggregator.java [hunk 1]
```diff
@@ -34,6 +34,11 @@
 
 public abstract class TermsAggregator extends DeferableBucketAggregator {
 
+    /**
+         * This class provides bucket thresholds configuration, but can be used ensure default value immutability
```
Comment (nik9000) [inline server/src/main/java/org/elasticsearch/search/aggregations/bucket/terms/TermsAggregator.java:38]: Sneaky extra indent

**Review by nik9000 — APPROVED** [review]

File: server/src/main/java/org/elasticsearch/search/aggregations/bucket/terms/TermsAggregator.java (hunk 1 above)
Comment (benwtrent) [reply server/src/main/java/org/elasticsearch/search/aggregations/bucket/terms/TermsAggregator.java:38]: literally unmergeable.

File: server/src/main/java/org/elasticsearch/search/aggregations/bucket/terms/TermsAggregator.java [hunk 2]
```diff
@@ -34,6 +34,11 @@
...
+         */
```
Comment (benwtrent) [inline server/src/main/java/org/elasticsearch/search/aggregations/bucket/terms/TermsAggregator.java:39]:
```suggestion
    /**
     * This class provides bucket thresholds configuration, but can be used ensure default value immutability
     */
```

**Comment by benwtrent** [comment]
@elasticmachine update branch

**Comment by benwtrent** [comment]
run elasticsearch-ci/part-1

**Pull request merged**